util/patricia-trie-ethereum/src/rlp_node_codec.rs

```diff
@@ -71,32 +108,38 @@ impl NodeCodec<KeccakHasher> for RlpNodeCodec<KeccakHasher> {
 		}
 	}
 	fn try_decode_hash(data: &[u8]) -> Option<<KeccakHasher as Hasher>::Out> {
-		let r = Rlp::new(data);
-		if r.is_data() && r.size() == KeccakHasher::LENGTH {
-			Some(r.as_val().expect("Hash is the correct size; qed"))
+
+		if data.len() == KeccakHasher::LENGTH {
+			let mut r = <KeccakHasher as Hasher>::Out::default();
+			r.as_mut().copy_from_slice(data);
+			Some(r)
 		} else {
 			None
 		}
+
 	}
 	fn is_empty_node(data: &[u8]) -> bool {
 		Rlp::new(data).is_empty()
 	}
-	fn empty_node() -> Vec<u8> {
-		let mut stream = RlpStream::new();
-		stream.append_empty_data();
-		stream.drain()
+
+	fn empty_node() -> &'static[u8] {
+		&[0x80]
 	}
 
-	fn leaf_node(partial: &[u8], value: &[u8]) -> Vec<u8> {
+	fn leaf_node(partial: Partial, value: &[u8]) -> Vec<u8> {
 		let mut stream = RlpStream::new_list(2);
-		stream.append(&partial);
+		stream.append_iter(encode_partial(partial, true));
 		stream.append(&value);
 		stream.drain()
 	}
 
-	fn ext_node(partial: &[u8], child_ref: ChildReference<<KeccakHasher as Hasher>::Out>) -> Vec<u8> {
+	fn extension_node(
+		partial: impl Iterator<Item = u8>,
+		nb_nibble: usize,
```

Comment:
could you apply the same renaming here as you did in substrate, e.g. `s/nb/number`, etc?